Extensions/BBText/JsExtension.js

```diff
@@ -0,0 +1,458 @@
+          .setLabel(_('Base color'))
+      );
+
+      objectProperties.set(
+        'opacity',
+        new gd.PropertyDescriptor(objectContent.opacity.toString())
+          .setType('number')
+          .setLabel(_('Opacity (0-255)'))
+      );
+
+      objectProperties.set(
+        'fontSize',
+        new gd.PropertyDescriptor(objectContent.fontSize)
+          .setType('number')
+          .setLabel(_('Base size'))
+      );
+
+      objectProperties.set(
+        'align',
+        new gd.PropertyDescriptor(objectContent.align)
+          .setType('choice')
+          .addExtraInfo('left')
+          .addExtraInfo('center')
+          .addExtraInfo('right')
+          .setLabel(_('Base alignment'))
+      );
+
+      objectProperties.set(
+        'fontFamily',
+        new gd.PropertyDescriptor(objectContent.fontFamily)
+          .setType('string')
+          .setLabel(_('Base font family'))
+      );
+
+      objectProperties.set(
+        'visible',
+        new gd.PropertyDescriptor(objectContent.visible ? 'true' : 'false')
+          .setType('boolean')
+          .setLabel(_('Visible on start'))
+      );
+
+      return objectProperties;
+    };
+    objectBBText.setRawJSONContent(
+      JSON.stringify({
+        text:
+          '[b]bold[/b] [i]italic[/i] [size=15]smaller[/size] [font=times]times[/font] font\n[spacing=12]spaced out[/spacing]\n[outline=yellow]outlined[/outline] [shadow=red]DropShadow[/shadow] ',
+        opacity: 255,
+        fontSize: '20',
+        visible: true,
+        color: '#000000',
+        fontFamily: 'Arial',
+        align: 'left',
+      })
+    );
+
+    objectBBText.updateInitialInstanceProperty = function(
+      objectContent,
+      instance,
+      propertyName,
+      newValue,
+      project,
+      layout
+    ) {
+      return false;
+    };
+    objectBBText.getInitialInstanceProperties = function(
+      content,
+      instance,
+      project,
+      layout
+    ) {
+      var instanceProperties = new gd.MapStringPropertyDescriptor();
+      return instanceProperties;
+    };
+
+    const object = extension
+      .addObject(
+        'BBText',
+        _('BBText'),
+        _(
+          'Displays a rich text label using BBCode markup (allowing to set parts of the text as bold, italic, use different colors and shadows).'
+        ),
+        'JsPlatform/Extensions/bbcode32.png',
+        objectBBText
+      )
+      .setIncludeFile('Extensions/BBText/bbtextruntimeobject.js')
+      .addIncludeFile('Extensions/BBText/bbtextruntimeobject-pixi-renderer.js')
+      .addIncludeFile('Extensions/BBText/dist/pixi-multistyle-text.umd.js');
+
+    // Single array approach to add all Setters/Getters, so we don't have to deal with SO much boilerplate repetition
+    // Could be useful for other extensions... Thinking of making an online tool to generate this file from json data some day
+    // 0 function name, 1 description, 2 icon name, 3 param property type, 4 hasModifier
```

Comment:
This comment can be removed if instead of arrays, you use objects :) 

```
{
  functionName: 
  description:
  iconName:
  type:
  hasModifier: // (Not sure what this is, maybe find another name?)
}
```

Reply:
the last one added the +-= modifier, so instead of just being able to set,, but I removed it and used the type of parameter it is to drive that

Reply:
Sounds better 👍